packages/shelter/src/util.ts

```diff
@@ -86,6 +86,65 @@ export function createSubscription<TState, TStoreData = any>(
   return data;
 }
 
+
+const storeInitPromises = new WeakMap<FluxStore, Promise<void>>();
+
+// awaits until the _isInitialized property of a store is true by overwritting it's setter
+function awaitStoreInit(store: FluxStore): Promise<void> {
+  return storeInitPromises[store] ?? (storeInitPromises[store] = new Promise<void>(resolve => {
+    if (store._isInitialized) resolve();
+
+    let actualIsInitialized = false;
+
+    Object.defineProperty("_isInitialized", store, {
+      get() {
+        return actualIsInitialized;
+      },
+      set(value) {
+        actualIsInitialized = value;
+        if (value === true) {
+          resolve();
+        }
+      }
+    })
+  }))
+}
+
+type storeCb = (store: FluxStore | FluxStore[]) => void
```

Comment:
```suggestion
type StoreCb = (store: FluxStore | FluxStore[]) => void
```
I'm being pedantic, but TypeScript type names are almost always [PascalCase](https://en.wikipedia.org/wiki/Camel_case) not camelCase, by convention